src/Apps/ArtAdvisor/07-Curated-Discovery/Components/Form/Form.tsx

```diff
@@ -0,0 +1,97 @@
+import {
+  Box,
+  Button,
+  Flex,
+  Join,
+  Spacer,
+  Spinner,
+  Text,
+  useToasts,
+} from "@artsy/palette"
+import { Interests } from "./Interests"
+import { Goals } from "./Goals"
+import { Budget } from "./Budget"
+import {
+  Action,
+  BudgetIntent,
+  State,
+} from "Apps/ArtAdvisor/07-Curated-Discovery/App"
+import { useState } from "react"
+
+interface FormProps {
+  state: State
+  dispatch: React.Dispatch<Action>
+}
+
+export const Form: React.FC<FormProps> = props => {
+  const [isLoading, setIsLoading] = useState(false)
+  const { state, dispatch } = props
+  const { sendToast } = useToasts()
+
+  const handleSubmit = async () => {
+    try {
+      setIsLoading(true)
+      const budgetIntent = await fetch("/api/advisor/7/budget/intent", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          budget: state.budget,
+        }),
+      })
+
+      if (budgetIntent.ok) {
+        const intent = (await budgetIntent.json()) as BudgetIntent
+        dispatch({ type: "SET_BUDGET_INTENT", intent })
+        dispatch({ type: "SET_STEP", step: "result" })
+      } else {
```

Comment:
Happy path: update the budget intent in the global state, and advance to the next (`Result`) screen.